docs/reference/modules/scripting/security.asciidoc

```diff
@@ -1,67 +1,62 @@
-
-The `script.*` settings allow for <<security-script-fine,fine-grained>>
-control of which script languages (e.g `painless`) are allowed to
-run in which context ( e.g. `search`, `aggs`, `update`), and where the script
-source is allowed to come from (i.e. `inline`, `stored`, `file`).
-
-For instance, the following setting enables `stored` `update` scripts for
-`painless`:
-
-[source,yaml]
-----------------
-script.engine.painless.inline.update: true
-----------------
-
-Less fine-grained settings exist which allow you to enable or disable scripts
-for all sources, all languages, or all contexts.  The following settings
-enable `inline` and `stored` scripts for all languages in all contexts:
-
-[source,yaml]
------------------------------------
-script.inline: true
-script.stored: true
------------------------------------
-
-WARNING:  The above settings mean that anybody who can send requests to your
-Elasticsearch instance can run whatever scripts they choose! This is a
-security risk and may well lead to your Elasticsearch cluster being
-compromised.
+While we make every effort to prevent scrips from running amok, security is
+something you do in
+https://en.wikipedia.org/wiki/Defense_in_depth_(computing)[layers], so you'll
+want to take some steps as well.
+
+First and foremost, never run Elasticsearch as the `root` user, as this would
+allow a script to access or do *anything* on your server, without limitations.
+Elasticsearch checks if it is running as `root` on startup but it won't
+refuse to start if it can't be sure so do your part to make sure it isn't
+running as `root`.
+
+You should not expose Elasticsearch directly to users, but instead have an
+application in between your users and Elasticsearch. If you *must* allow users
+the ability to run whatever request they want then make sure that dynamic
+scripting is disabled for all non-sandboxed languages. That is already the
+default. Even then, understand that it is possible for a sufficiently
+determined malicious user to write searches that overwhelm the Elasticsearch
+cluster and bring it down.
+
+By default Elasticsearch enables dynamic scripting for sandboxed languages,
+namely the scripting language `painless`, the template language `mustache`, and
+the expression language `expressions`. These *ought* to be safe to expose to
+trusted users and to your application servers. It is tempting fate to expose
+them to the Internet at large, on the other hand. Your applications can use
+these scripts dynamically and be exposed to the Internet. Just don't let the
+Internet write scripts.
+
+Enabling dynamic scripting for non-sandboxed languages like `groovy`, `python`,
+and `javascript` is super dangerous. These languages have unfettered access to
+Elasticsearch and can cause all kind of trouble. Worse, the only thing
+preventing them from causing real trouble is the Java security manager and
+SECCOMP/Seatbelt/ActiveProcessLimit. Assuming that those will prevent disaster
```

Comment:
I don't like this line starting "Assuming..." - it sounds like you're saying that the security manager etc is pretty much a waste of time.

I'd have a section explaining the other layers of security that we have such as JSM, seccomp, etc.  Then say something like "Even with those in place, non-sandboxed languages provide the malicious user with too many opportunities to attack your cluster or to mount a denial of service attack"